tests/torch/test_prepare_for_inference.py

```diff
@@ -0,0 +1,345 @@
+
+
+def _get_config_for_algo(algo_names, input_size, quant_mode='symmetric', overflow_fix="enable"):
+    config = NNCFConfig()
+    config.update({"model": "model", "input_info": {"sample_size": input_size}, "compression": []})
+
+    if "quantization" in algo_names:
+        config["compression"].append(
+            {
+                "algorithm": "quantization",
+                "initializer": {"range": {"num_init_samples": 0}},
+                "weights": {"mode": quant_mode},
+                "activations": {"mode": quant_mode},
+                "overflow_fix": overflow_fix,
+            }
+        )
+    if "filter_pruning" in algo_names:
+        config["compression"].append(
+            {
+                "algorithm": "filter_pruning",
+                "params": {"schedule": "baseline", "num_init_steps": 1},
+                "pruning_init": 0.5,
+            }
+        )
+
+    return config
+
+
+def _gen_input_tensor(shape):
+    n = np.prod(list(shape))
+    res = torch.Tensor(range(n)) / torch.Tensor([n - 1]) * torch.Tensor([2]) - torch.Tensor([1.0])
+    res[n // 2] = 0.0
+    return res.reshape(shape)
+
+
+def _check_operators(model):
+    """Check that model contains only 8bit FakeQuantize operators."""
+
+    if hasattr(model, "external_quantizers"):
+        for key in list(model.external_quantizers.keys()):
+            op = model.external_quantizers[key]
+            assert isinstance(model.external_quantizers[key], FakeQuantize)
+            assert op.quant_max - op.quant_min == 255
+
+    for node in model.get_original_graph().get_all_nodes():
+        if node.node_type in ["nncf_model_input", "nncf_model_output"]:
+            continue
+
+        nncf_module = model.get_containing_module(node.node_name)
+
+        if hasattr(nncf_module, "pre_ops"):
+            for key in list(nncf_module.pre_ops.keys()):
+                op = nncf_module.get_pre_op(key).op
+                assert isinstance(op, FakeQuantize)
+                assert op.quant_max - op.quant_min == 255
+
+        if hasattr(nncf_module, "post_ops"):
+            for key in list(nncf_module.post_ops.keys()):
+                op = nncf_module.post_ops(key).op
+                assert isinstance(op, FakeQuantize)
+                assert op.quant_max - op.quant_min == 255
+
+
+def _idfn(val):
+    if isinstance(val, tuple):
+        return "{}".format("-".join([str(v) for v in val]))
+    return None
+
+
+INPUT_TEST_SCALES = (
+    (1, 16, 24, 24),
+    (16, 8, 12, 12),
+)
+
+
+@pytest.mark.parametrize("use_cuda", (True, False), ids=("cuda", "cpu"))
+@pytest.mark.parametrize("half_range", [False, True], ids=["full_range", "half_range"])
+@pytest.mark.parametrize("is_weights", (True, False), ids=("weights", "activation"))
+@pytest.mark.parametrize("is_signed", (True, False), ids=("signed", "unsigned"))
+@pytest.mark.parametrize("input_size", INPUT_TEST_SCALES, ids=_idfn)
+@pytest.mark.parametrize("is_per_channel", (True, False), ids=("per_channel", "per_tensor"))
```

Comment:
Do you think that you could reuse this parametrization for the regular NNCF FQs as well so as not to duplicate code and be sure that torch FQs are validated at the same parameter space as NNCF FQs? You can set up a `pytest.fixture` for that and reuse it both in `test_prepare_for_inference.py` and `test_functions.py`

Reply:
Good idea. But i think better do it in new PR, as this requires a change in non-PR related tests, and can be reused for another tests not only in `test_prepare_for_inference.py` and `test_functions.py`.
But it can changes name of current tests (order of parametrized arguments and ids). It will not be a problem, is it? 

To be clear, do you suggest?

Add to contest.py fixtures for `use_cuda`, `half_range`, `is_weights`, `is_per_channel`, `is_signed`:

```
@pytest.fixture(params=[True, False], ids=["per_channel", "per_tensor"])
def is_per_channel(request):
    return request.param
```

Remove parametrize from tests:

`@pytest.mark.parametrize("is_weights", (True, False), ids=("weights", "activation"))
`